app/jobs/set_catkeys_and_barcodes_job.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+##
+# Job to update/add catkey/barcodes to objects
+class SetCatkeysAndBarcodesJob < GenericJob
+  attr_reader :catkeys, :barcodes
+
+  ##
+  # A job that allows a user to specify a list of pids and a list of catkeys to be associated with these pids
+  # @param [Integer] bulk_action_id GlobalID for a BulkAction object
+  # @param [Hash] params additional parameters that an Argo job may need
+  # @option params [Array] :pids required list of
+  # @option params [String] :catkeys list of catkeys to be associated 1:1 with pids in order
+  # @option params [String] :catkeys_option option to update the catkeys
+  # @option params [String] :barcodes list of barcodes to be associated 1:1 with pids in order
+  # @option params [String] :barcodes_option option to update the barcodes
+  # @option params [String] :csv_file CSV string
+  def perform(bulk_action_id, params)
+    super
+
+    update_pids, catkeys, barcodes = params_from(params)
+
+    with_bulk_action_log do |log|
+      log.puts("#{Time.current} Starting ManageCatkeyJob for BulkAction #{bulk_action_id}")
+      update_druid_count
+      update_pids.each_with_index do |current_druid, i|
+        change_set = ItemChangeSet.new do |change|
+          change.catkey = catkeys[i] if catkeys
+          change.barcode = barcodes[i] if barcodes
```

Comment:
Should these conditions be checking e.g. `catkeys` or `catkeys[i]`?

Reply:
`catkeys`. A null indicates that catkeys was not selected for updating.

Reply:
@justinlittman Ah, I see, it uses the `use_*_option` params (unless CSV file used instead, in which case the other args are ignored). Thanks.